lib/ok.ex

```diff
@@ -150,6 +150,62 @@ defmodule OK do
   def is_failure?({:ok, _value}), do: false
   def is_failure?({:error, _reason}), do: true
 
+  @doc guard: true
+  @doc """
+  Checks if a result tuple is tagged as `:error`, and returns `true` if so.
+  If the tuple is tagged as `:ok`, returns `false`.
+
+  Allowed in guards.
+
+  ## Examples
+
+      iex> require OK
+      ...> f = fn result when OK.is_success(result) -> "ok" end
+      ...> f.({:ok, "some value"})
+      "ok"
+
+      iex> require OK
+      ...> f = fn result when OK.is_success(result) -> "ok" end
+      ...> f.({:error, :some_reason})
+      ** (FunctionClauseError) no function clause matching in anonymous fn/1 in OKTest.\"doctest OK.is_success/1 (31)\"/1
+
+      iex> require OK
+      ...> f = fn result when OK.is_success(result) -> "ok" end
+      ...> f.(nil)
+      ** (FunctionClauseError) no function clause matching in anonymous fn/1 in OKTest.\"doctest OK.is_success/1 (32)\"/1
+  """
+  @spec is_success(term()) :: Macro.t()
```

Comment:
Do you know why the return value of this in not boolean?

Reply:
AFAIK, `defguard` returns a macro.  Dialyzer was happy with it.

https://hexdocs.pm/elixir/Kernel.html#defguard/1

With boolean
```
lib/ok.ex:177:invalid_contract
Invalid type specification for function.

Function:
OK.MACRO-is_success/2

Success typing:
@spec MACRO-is_success({_, %{:line => _, _ => _}} | map(), _) ::
  {:__block__ | {:., [], [:andalso | :erlang, ...]}, [],
   [{:= | {_, _, _}, [], [any(), ...]}, ...]}
________________________________________________________________________________
lib/ok.ex:205:invalid_contract
Invalid type specification for function.

Function:
OK.MACRO-is_failure/2

Success typing:
@spec MACRO-is_failure({_, %{:line => _, _ => _}} | map(), _) ::
  {:__block__ | {:., [], [:andalso | :erlang, ...]}, [],
   [{:= | {_, _, _}, [], [any(), ...]}, ...]}
________________________________________________________________________________
```

Reply:
I wonder if these are even valuable as specs, takes anything and returns code doesn't give much information, but  if dialyzer is happy then it works for me.